osutil/kcmdline.go

```diff
@@ -204,6 +204,115 @@ func KernelCommandLineKeyValues(keys ...string) (map[string]string, error) {
 	return m, nil
 }
 
+// KernelValidateParam validates a kernel parameter and splits it in
+// key and value in case there is no error.
+// See lib/cmdline.c:next_arg in kernel sources.
+
+// KernelArgument represents a parsed kernel argument
+type KernelArgument struct {
+	Param string
+	Value string
+}
+
+// KernelParseCommandline parses a kernel command line, returning a
+// slice with the arguments in the same order as in cmdline. Note that
+// kernel arguments can be repeated. We follow the same algorithm as in
+// the kernel as far as possible.
```

Comment:
(nitpick) Is there also a kernel code reference that we could use here? If so, maybe add `Follow the linux kernels  lib/cmdline.c:next_arg as closely as possible` (I know it's slightly redundant as it's also mentioned a few lines above but I think that is worth it)